### PR TITLE
[RHPAM-1488] Add JDG indexing support to OpenShift HA Authoring in RHPAM

### PR DESCRIPTION
**Pull request by errantepiphany**
[RHPAM-1488] Add JDG indexing support to OpenShift HA Authoring in RHPAM
https://issues.jboss.org/browse/RHPAM-1488

Signed-off-by: David Ward <dward@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by sutaakar** [comment]
@errantepiphany Is this new template fully supported? Or is it a tech preview?

**Comment by errantepiphany** [comment]
> @errantepiphany Is this new template fully supported? Or is it a tech preview?

@sutaakar Let's discuss in our meeting, but 99% sure it's going to be tech preview. I had to build the datagrid 7.3 image myself since it's not even in the customer catalog...

**Comment by errantepiphany** [comment]
I think the validator tool (executed by circleci) is getting confused on the stateful sets...

**Comment by errantepiphany** [comment]
Note that RHPAM-1488 is followed up on by RHPAM-1924 because of AF-1868.

**Comment by spolti** [comment]
Validator fix is available here: https://github.com/jboss-container-images/jboss-kie-modules/pull/164

File: templates/rhpam73-authoring-ha-trial-datagrid.yaml [hunk 1]
```diff
@@ -0,0 +1,1799 @@
...
+                  {
+                    "name": "${APPLICATION_NAME}-amq-pvol",
+                    "mountPath": "/opt/amq/data"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+    creationTimestamp: null
```
Comment (spolti) [inline templates/rhpam73-authoring-ha-trial-datagrid.yaml]: can be removed,.

File: templates/rhpam73-authoring-ha-trial-datagrid.yaml [hunk 2]
```diff
@@ -0,0 +1,1799 @@
...
+    podManagementPolicy: OrderedReady
+    replicas: 2
+    revisionHistoryLimit: 10
+    selector:
+      matchLabels:
+        deploymentConfig: ${APPLICATION_NAME}-amq
+    serviceName: ${APPLICATION_NAME}-amq-tcp
+    template:
+      metadata:
+        creationTimestamp: null
```
Comment (spolti) [inline templates/rhpam73-authoring-ha-trial-datagrid.yaml]: can be removed.

File: templates/rhpam73-authoring-ha-trial-datagrid.yaml [hunk 3]
```diff
@@ -0,0 +1,1799 @@
...
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        schedulerName: default-scheduler
+        securityContext: {}
+        terminationGracePeriodSeconds: 60
+    updateStrategy:
+      type: OnDelete
+    volumeClaimTemplates:
+    - metadata:
+        creationTimestamp: null
```
Comment (spolti) [inline templates/rhpam73-authoring-ha-trial-datagrid.yaml]: can be removed.

**Comment by errantepiphany** [comment]
Closing to recreate with more descriptive branch for 7.3.x.

**Pull request closed without merging**